misc/python/materialize/bump-main-version.py

```diff
@@ -0,0 +1,34 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+"""Cut a new release and push it to the Materialize repo"""
+
+import sys
+
+from materialize import MZ_ROOT, spawn
+from materialize.git import get_latest_version, get_remote, create_branch
+from materialize.mz_version import MzVersion
+
+
+def main():
+    latest_version = get_latest_version(version_type=MzVersion)
+    version = str(latest_version.bump_minor().replace(prerelease = 'dev'))
+    print(f"Creating branch bump-version")
+    create_branch("bump-version")
+    print(f"Bumping version on main to {version}")
+    spawn.runv([MZ_ROOT / "bin" / "bump-version", version])
+    remote = get_remote()
+    print(f"Pushing branch 'bump-version' to {remote}")
+    spawn.runv(["git", "push", remote])
+    with open("main-version.txt", "w") as f:
+        f.write(version)
```

Comment:
The more modern way to write this is like so:

```suggestion
    Path("main-version.txt").write_text(version)
```

Reply:
It just occurred to me that we can integrate with GitHub Action's support for outputs here: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/passing-information-between-jobs#example-defining-outputs-for-a-job

```suggestion
    github_output = os.getenv("GITHUB_OUTPUT")
    if github_output:
        Path(github_output).write_text(f"version={version}")
```

Reply:
Let's hope it works! This part is totally untested.